manim/mobject/mobject.py

```diff
@@ -2617,6 +2624,18 @@ def construct(self):
                     circ.become(square)
                     self.wait(0.5)
         """
+        # changes the new mobject's center, height, or width to the original's before transformation
+        # if both width and height are matched then the object is stretched to fit both
+        if match_height and match_width:
```

Comment:
Without having tested it: is this separate clause really necessary? If so, what is the difference to
```py
if match_height:
    mobject.match_height(self)
if match_width:
    mobject.match_width(self)
```

Reply:
I did that because the behavior changes depending on the order they are in. I didn't want one to take precedence, so I had it stretch to fit both. Otherwise, having both enabled would only match width. It's kind of arbitrary, but I felt like it made the most sense. I can always remove it.

Reply:
Hmm. I'm not sure I like this behavior; it sort of seems as if the software is trying to be smarter than its user.

Counter-suggestion: what if we introduced another keyword argument; `stretch` (default False) with an implementation like
```py
if stretch:
    mobject.stretch_to_fit_height(...)
    mobject.stretch_to_fit_width(...)
    mobject.stretch_to_fit_depth(...)
else:
    if match_height:
        mobject.match_height(self)
    if match_width:
        mobject.match_width(self)
    if match_depth:
        mobject.match_depth(self)

if match_center:
    mobject.move_to(self.get_center())
```

Reply:
done! let me know if there's anything else I missed